src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptionsConfigExtensions.cs

```diff
@@ -48,6 +48,9 @@ public static class MessageDrivenSubscriptionsConfigExtensions
         {
             Guard.AgainstNullAndEmpty(nameof(publisherEndpoint), publisherEndpoint);
 
+            if (publisherEndpoint.Contains("@"))
+                throw new ArgumentException($"Expected an endpoint name but received '{publisherEndpoint}'.");
```

Comment:
also, need to add those checks to the assembly overloads (probably need to rebase first)